tests/test_numpy_dtypes.cpp

```diff
@@ -251,6 +251,38 @@ py::list test_dtype_ctors() {
     return list;
 }
 
+py::list test_dtype_kind() {
+    py::list list;
+    for (auto& dt : {
+                     py::dtype("bool8"),   // bool
+                     py::dtype("int16"),   // short
+                     py::dtype("int32"),   // int
+                     py::dtype("int64"),   // long int
+                     py::dtype("float32"), // float
+                     py::dtype("float64"), // double
+                     py::dtype("float128") // long double
+                    }) {
+        list.append(dt.kind());
+    }
+    return list;
+}
+
+py::list test_dtype_char_() {
+    py::list list;
+    for (auto& dt : {
+                     py::dtype("bool8"),   // bool
+                     py::dtype("int16"),   // short
+                     py::dtype("int32"),   // int
+                     py::dtype("int64"),   // long int
+                     py::dtype("float32"), // float
+                     py::dtype("float64"), // double
+                     py::dtype("float128") // long double
```

Comment:
On Winblows `sizeof(long double) == sizeof(double)`, which might explain the error.

Reply:
I was about to say: MSVC doesn't really like this, @bemichel.

Reply:
Also, those comments are not completely right: sometimes `np.int32 == np.int`, but sometimes `np.int64 == np.int`. It really depends on the platform, I think?

But then, as far as I remember (I once contributed to this part of NumPy's docstrings in a sprint at a conference), the letters are actually linked to `np.bool`, `np.int`, np.float`, etc, and not to the sized variants!

~Scratch that. Seems that either they changed, or I misremember.~

No, I was right. But `np.int` is an alias for a Python `int`, not for a C int, that's `np.intc`.

Reply:
At any rate, here are the docs:
https://numpy.org/doc/stable/reference/arrays.scalars.html#built-in-scalar-types

`intN` or `floatN` are aliases, so I'd suggest using the non-alias versions.

Reply:
Actually. It's easier if I just try fixing this. Let's see.

Reply:
Done. Let's see what this says and does.